ttnn/cpp/ttnn/operations/data_movement/concat/concat.cpp

```diff
@@ -87,6 +89,17 @@ MassagedConcat build_unsqueeze_concat(int input_rank, const MemoryConfig& output
 }
 
 MassagedConcat build_untilize_rm_retilize_concat(uint8_t queue_id, const MemoryConfig& output_memory_config) {
+    auto compute_output_shape = [](const std::vector<ttnn::Tensor>& tensors, int dim) -> ttnn::SimpleShape {
+        ttnn::SimpleShape shape_out = tensors[0].get_logical_shape();
+        shape_out[dim] = 0;
+        for (const Tensor& in_ref : tensors) {
+            ttnn::SimpleShape curr_shape = in_ref.get_logical_shape();
+            shape_out[dim] += curr_shape[dim];
+        }
+        return shape_out;
+    };
+
+    std::shared_ptr<ttnn::SimpleShape> logical_output_shape = std::make_shared<ttnn::SimpleShape>();
```

Comment:
shared_ptr? wow wow, why?

Reply:
The lambdas capturing this are returned from this call so we need the shared_ptr to avoid dangling stack references. I'm open to other ideas :)

Reply:
From what I see, this code is in `build_untilize_rm_retilize_concat`, which is only called from `ConcatOperation::invoke`, which actually has `tensors` and `dims` parameters. This means you can add more parameters to `build_untilize_rm_retilize_concat` and make a call to `compute_output_shape` right on this line, which would avoid this sharing by shared_ptr

Reply:
Stas comment makes sense to me

Reply:
Okay, I've made the change. @ntarafdar do you mind babysitting the jobs I kick off in a second? I'll link them in the body of the PR as usual.

@ayerofieiev-tt Going forward, can we keep micro-optimization suggestions like this as "approve with comments"? With how flaky CI is at the moment, I think that any perf optimization that doesn't register on models perf regressions should be reserved for future PRs. I agree with the concerns raised but I don't think they're critical enough to hold back a P0 fix.

Reply:
tldr, sure, going forward but case by case 👍 
———
To be clear, this is not a performance optimization. The code smelled and required a correction. I saw no reason to merge it as is because it looked like a small fix that better to do immediately. When we postpone things like this (small on the scale), there is a risk to never resolve the issue, leaving it for the next generation to replicate. Please understand my concern. 

Leaving approve before comments are addressed is something we all desire as it is a sign of trust that the feedback will be reasonably resolved and not showed away “for later”. 

Agree?